### PR TITLE
Group add automation elements in dialog

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Split conditions and action in building blocks.
Group trigger/conditions/action in sections in a dialog when adding a new one.

CleanShot 2023-12-19 at 11 48 17

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by matthiasdebaat** [comment]
What do you think about adding a diver line above `other`?

File: src/panels/config/automation/action/ha-automation-action.ts [hunk 1]
```diff
@@ -213,6 +165,43 @@ export default class HaAutomationAction extends LitElement {
     }
   }
 
+  private _addActionDialog() {
+    showAddAutomationElementDialog(this, {
+      type: "action",
+      add: this._addAction,
+      clipboardItem: getType(this._clipboard!.action),
```
Comment (piitaya) [inline src/panels/config/automation/action/ha-automation-action.ts:172]: Clipboard can be undefined. We should do that every clipboard is used.
```suggestion
      clipboardItem: getType(this._clipboard?.action),
```

File: src/panels/config/automation/add-automation-element-dialog.ts [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+          rootTabbable
+          dialogInitialFocus
+        >
+          ${this._params.clipboardItem &&
+          !this._filter &&
+          (!this._group ||
+            items.find((item) => item.key === this._params!.clipboardItem))
+            ? html`<ha-list-item
+                twoline
+                activated
```
Comment (piitaya) [inline src/panels/config/automation/add-automation-element-dialog.ts:434]: Why is it activated?
Should like a bug in the dialog as we already have hover state on each item.

CleanShot 2023-12-20 at 09 39 25

Comment (matthiasdebaat) [reply src/panels/config/automation/add-automation-element-dialog.ts:434]: Let's use a divider

Comment (matthiasdebaat) [reply src/panels/config/automation/add-automation-element-dialog.ts:434]: And in the long run add a Toolbar to the automation editor

**Review by piitaya — APPROVED** [review]
LGTM ! 👍

**Pull request merged**